### PR TITLE
Group node

### PR DESCRIPTION
**Pull request by martint-unity**
### Purpose of this PR
Adding the long lost son of shader graph; Group Node.
With this you can now finally group your nodes and keep your Graphs organized and nice.

---
### Release Notes

---
### Testing status
Running Katana
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders/proj57-All%20Tests/builds/1365?automation-tools_branch=add-platform-filter&ScriptableRenderLoop_branch=sg/group-node&unity_branch=trunk

---
### Overall Product Risks
Medium

**Halo Effect**: None, Low, Medium, High?
Medium

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by zadi0648** [comment]
Copying & pasting a group doesn't work. Also if you copy and paste, then undo and redo, then it will be a group. But that bug shouldn't matter if we make it so that it creates a group on the first copy and paste.
copypastegroupundoredobug

**Comment by zadi0648** [comment]
VFX copies & paste the entire group so this is a parity thing.
vfxcopyandpaste

**Comment by zadi0648** [comment]
This is not a big deal and can be excused but the default sub node thing (what is it called?) sticks out of the group. Is that a simple fix or a graph view fix?
nodedefaultsticksout

**Comment by zadi0648** [comment]
I believe otherwise this is good. I have a couple of concerns with the node grouping, but they reproduce in vfx so I create some suggestions for graph view team here: https://unity.slack.com/archives/C5196JW2E/p1542246325016900

**Comment by zadi0648** [comment]
**Summary:** Tested by creating new Shader Graphs as well as trying out grouping nodes in prexisting shader graph files. I also tested against VFX to clarify the difference between sg only bugs / concerns and graph view wide bugs / concerns. I found some bugs / concerns with graph view's group node which are posted here: https://unity.slack.com/archives/C9TAPHG06/p1542250727035700. Currently the only existing problem is that default value subnodes stick outside of the group which can be marked fix-later as it's not a major problem.

✖ Confirming that all nodes fit inside the group (currenlty default value subnodes do not - will approve and mark as fix-later)
✔ Moving groups around randomly & nodes inside 
✔ Overlapping groups
✔ Delete Nodes inside the group
✔ Twirling nodes around and doing crazy things with them to mess with the size
✔ Ungrouping apart of a group
✔ Ungrouping an entire group
✔ Group different kinds of things, including properties, inlines, & sub graphs
✔ Undo / Redoing all of the above (bug found for copying & pasting - fix confirmed)
✖ Testing if group node changes triggers file-save-needed (doesn't but Matt's fixes aren't in yet)

Creating new Shader Graphs and using Node Grouping:
newgraphcapture0

Create Groups for Existing Shader Graphs:
oldgraphcapture0

**Comment by Kink3d** [comment]
@martint-unity this needs documentation. Feel free to merge the PR without it but please remember to add docs for this feature before 19.1 lands!

File: com.unity.shadergraph/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [5.1.0-preview] - 2018-10-30
 ### Added
 - You can now show and hide the Main Preview and the Blackboard from the toolbar.
+- LWRP now has __Group Node__, where you can group together several nodes. You can use this to keep your Graphs organized and nice.
```
Comment (pbbastian) [inline com.unity.shadergraph/CHANGELOG.md]: This feature is not specific to LWRP :)

Comment (pbbastian) [reply com.unity.shadergraph/CHANGELOG.md]: @Katstodian

**Pull request merged**